### DISCUSSION TITLE
pyrootutils alternative?

### DISCUSSION THREAD

**Discussion by cpoptic** [ashleve/lightning-hydra-template#437] [Q&A]
What is the purpose of the `pyrootutils` package in the context of lightning-hydra-template?  There are other projects that do not have to use such a 3rd party library.  
For example, https://github.com/Erlemar/pytorch_tempest does not make use of `pyrootutils`, and it works fine.  
Nor does https://github.com/Erlemar/wheat make use of pyrootutils.

Is there another way to get the project root without having to introduce a 3rd party package?

**Comment by ashleve** [comment]
Hi @cpoptic, this has been recently explained here with a comment:

https://github.com/ashleve/lightning-hydra-template/blob/2f3fe8642ff56a325cc053b32629947be3d322c9/src/train.py#L25-L28

https://github.com/ashleve/lightning-hydra-template/blob/2f3fe8642ff56a325cc053b32629947be3d322c9/src/train.py#L10-L31

To sum up, the purpose is to make the environment more robust and convenient.

You should be able to easily get rid of it as described above.

In the project you mentioned, nothing like that is used since the author stores all entry scripts in top-level directory. The problems are:
- in my opinion, it's not very clean to have entry files among other files like linter configs
- you can't nest your entry scripts in `src/` unless you install project as a package (forcing the user to install project is bothersome so we try to avoid that)
- If you try to run the script from any other place than project root, the logging and data paths will be different due to different working directory 

You are of course free to get rid of it if you find it too risky to have such dependency in your codebase.
